app/helpers/filters_helper.rb

```diff
@@ -309,13 +309,21 @@ def formatted_assigned_to_filter(session_filters)
     return "All" if session_filters["assigned_to"].include?("all")
     return "You" if session_filters["assigned_to"].include?("you")
 
-    User.own_and_managing_org_users(current_user.organisation).find(session_filters["user"].to_i).name
-    selected_user_option = User.own_and_managing_org_users(current_user.organisation).find(session_filters["user"].to_i)
+    selected_user_option = find_selected_specific_user(session_filters)
     return unless selected_user_option
 
     "#{selected_user_option.name} (#{selected_user_option.email})"
   end
 
+  def find_selected_specific_user(session_filters)
+    user_id = session_filters["user"].to_i
+    if current_user.support?
+      User.find(user_id)
+    else
+      User.own_and_managing_org_users(current_user.organisation).find(user_id)
+    end
+  end
```

Comment:
I'm wondering a couple of things (not really introduced by this PR, but relevant to the bug);

1. What set of users do we show people to select from for this filter? I would have expected any related org, not just own or managing orgs, but I might be wrong there. Should make sure that matches what we'll allow here.
2. Would it be sensible to set up a scope on the User model to use for this kind of restriction? (Where that scope was aware of whether the current user was support or not). That might actually be adjusting the current "visible" scope of whether a user a has been deleted, if we thought it made sense to tie that in - which it probably does.

Reply:
I'll check 1. and 2. is a good idea, will look into it, maybe keeping the existing visible scope and creating a new one "visible_to".

Reply:
(1) We are choosing the users to show people to select from in the same way already, so if support all users, if not, from their own or managing orgs.
(2) I'll the "visible_to_user" scope and amend all places, while keeping the logic of own or managing orgs. We can adjust if needed.